pkg/ec2helper/ec2helper.go

```diff
@@ -1333,3 +1306,187 @@ func HasEbsVolume(image *ec2.Image) bool {
+					}
+					blockDevices[index].SetEbs(blockDeviceEbs)
+				}
+			}
+			requestInstanceConfig.LaunchTemplateBlockMappings = blockDevices
+		}
+		setAutoTermination = IsLinux(*detailedConfig.Image.PlatformDetails) && simpleConfig.AutoTerminationTimerMinutes > 0
+	}
+
+	if setAutoTermination {
+		requestInstanceConfig.InstanceInitiatedShutdownBehavior = aws.String("terminate")
+		autoTermCmd := fmt.Sprintf("#!/bin/bash\necho \"sudo poweroff\" | at now + %d minutes\n",
+			simpleConfig.AutoTerminationTimerMinutes)
+		if simpleConfig.BootScriptFilePath == "" {
+			requestInstanceConfig.UserData = aws.String(base64.StdEncoding.EncodeToString([]byte(autoTermCmd)))
+		} else {
+			bootScriptRaw, _ := ioutil.ReadFile(simpleConfig.BootScriptFilePath)
+			bootScriptLines := strings.Split(string(bootScriptRaw), "\n")
+			//if #!/bin/bash is first, then replace first line otherwise, prepend termination
+			if len(bootScriptLines) >= 1 && bootScriptLines[0] == "#!/bin/bash" {
+				bootScriptLines[0] = autoTermCmd
+			} else {
+				bootScriptLines = append([]string{autoTermCmd}, bootScriptLines...)
+			}
+			bootScriptRaw = []byte(strings.Join(bootScriptLines, "\n"))
+			requestInstanceConfig.UserData = aws.String(base64.StdEncoding.EncodeToString(bootScriptRaw))
+		}
+	} else {
+		if simpleConfig.BootScriptFilePath != "" {
+			bootScriptRaw, _ := ioutil.ReadFile(simpleConfig.BootScriptFilePath)
+			requestInstanceConfig.UserData = aws.String(base64.StdEncoding.EncodeToString(bootScriptRaw))
+		}
+	}
+
+	return requestInstanceConfig
+}
+
+func (h *EC2Helper) DeleteLaunchTemplate(templateId *string) error {
+	fmt.Println("Deleting Launch Template...")
+	input := &ec2.DeleteLaunchTemplateInput{
+		LaunchTemplateId: templateId,
+	}
+
+	_, err := h.Svc.DeleteLaunchTemplate(input)
+	return err
+}
+
+func (h *EC2Helper) LaunchFleet(templateId *string) (*ec2.CreateFleetOutput, error) {
+	fleetTemplateSpecs := &ec2.FleetLaunchTemplateSpecificationRequest{
+		LaunchTemplateId: templateId,
+		Version:          aws.String("$Latest"),
+	}
+
+	fleetTemplateConfig := []*ec2.FleetLaunchTemplateConfigRequest{
+		{
+			LaunchTemplateSpecification: fleetTemplateSpecs,
+		},
+	}
+
+	spotRequest := &ec2.SpotOptionsRequest{
+		AllocationStrategy: aws.String("capacity-optimized"),
+	}
+
+	targetCapacity := &ec2.TargetCapacitySpecificationRequest{
+		DefaultTargetCapacityType: aws.String("spot"),
+		OnDemandTargetCapacity:    aws.Int64(0),
+		SpotTargetCapacity:        aws.Int64(1),
+		TotalTargetCapacity:       aws.Int64(1),
+	}
+
+	input := &ec2.CreateFleetInput{
+		LaunchTemplateConfigs:       fleetTemplateConfig,
+		SpotOptions:                 spotRequest,
+		TargetCapacitySpecification: targetCapacity,
+		Type:                        aws.String("instant"),
+	}
+
+	result, err := h.Svc.CreateFleet(input)
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			fmt.Println(aerr.Error())
+		} else {
+			fmt.Println(err.Error())
+		}
+		return nil, err
+	}
```

Comment:
Can you add the error handling code from your [downstream bubbletea PR](https://github.com/GavinBurris42/aws-simple-ec2-cli/pull/9) here as well? It allows us to surface CreateFleet call errors and indicate to the user why the instance launch failed.

```suggestion
	} else {
		if len(result.Errors) != 0 {
			err = errors.New(*result.Errors[0].ErrorMessage)
			cli.ShowError(err, "Creating spot instance failed")
			return nil, err
		}
	}
```